sort/topsort.py

```diff
@@ -1,60 +1,79 @@
-"""
-Given a list of system packages,
-some packages cannot be installed until the other packages are installed.
-Provide a valid sequence to install all of the packages.
+import unittest
 
-e.g.
-a relies on b
-b relies on c
+GRAY, BLACK = 0, 1
 
-then a valid sequence is [c, b, a]
-"""
+def topological_sort_recursive(graph):
+    order, enter, state = [], set(graph), {}
+    
+    def dfs(node):
+        state[node] = GRAY
+        #print(node)
+        for k in graph.get(node, ()):
+            sk = state.get(k, None)
+            if sk == GRAY: raise ValueError("cycle")
+            if sk == BLACK: continue
```

Comment:
Please avoid one liner conditional statements. Make them two lines for readability.